### ISSUE TITLE
Failed stopping the global hook

### ISSUE THREAD

**Issue by daltzctr** [TolikPylypchuk/SharpHook#48]
Sometimes, calling `Dispose` on the global hook fails with the following exception
```
SharpHook.HookException: Failed stopping the global hook: Failure (1)
   at SharpHook.GlobalHookBase.Dispose(Boolean disposing)
   at SharpHook.GlobalHookBase.Dispose()
   at MyApp.Dispose()
   at MyApp.Dispose()
   at MyApp.OnDisappearing()
   at Microsoft.Maui.Controls.Page.SendDisappearing()
   at Microsoft.Maui.Controls.ShellSection.OnPopAsync(Boolean animated)
   at Microsoft.Maui.Controls.ShellSection.GoToAsync(ShellNavigationRequest request, ShellRouteParameters queryData, IServiceProvider services, Nullable`1 animate, Boolean isRelativePopping)
   at Microsoft.Maui.Dispatching.DispatcherExtensions.<>c__DisplayClass3_0.<<DispatchAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.Maui.Dispatching.DispatcherExtensions.<>c__DisplayClass2_0`1.<<DispatchAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.Maui.Controls.ShellNavigationManager.GoToAsync(ShellNavigationParameters shellNavigationParameters, ShellNavigationRequest navigationRequest)
   at Microsoft.Maui.Controls.ShellSection.NavigationImpl.OnPopAsync(Boolean animated)
   at Microsoft.Maui.Controls.Shell.<>c__DisplayClass265_0.<<OnBackButtonPressed>g__NavigationPop|0>d.MoveNext()

```

**Comment by TolikPylypchuk** [comment]
Hi! Thanks for creating an issue!

Could you please provide the [libuiohook debug logs](https://sharphook.tolik.io/v4.2.0/articles/logging.html) that are output when this exception happens? This will help troubleshoot the issue.

**Comment by daltzctr** [comment]
@TolikPylypchuk 

This does not ever get triggered. Additionally, when it does crash. Mouse control becomes extremely laggy for about ~3seconds, with no obvious rise in CPU usage (or other stats shown in Task Manager). I've confirmed that the following code does actually register the event, the event just never gets triggered.

```
        // Set logging for libuihook
        var logSource = LogSource.Register(minLevel: SharpHook.Native.LogLevel.Info);
        logSource.MessageLogged += (s, e) =>
        {
            Log.Information(e.LogEntry.FullText);
        };
```

**Comment by TolikPylypchuk** [comment]
Mouse lagging is also caused by breakpoints in Visual Studio (as described in issue #37) - I think it may be similar.

As for the logs - could you please set the minimum level to `Debug`? libuiohook has much more debug logs than any other log levels - it actually logs every single input event, and I'm sure stopping the hook will also get something logged.

**Comment by TolikPylypchuk** [comment]
Hi @cmsw97! Thanks for the example code! I was able to reproduce the issue and I think that this line is particularly suspicious:

```
hook_run [816]: Create invisible window failed! (0X582)
```

I'll investigate what causes the issue and hopefully fix it in the nearest future.

**Comment by TolikPylypchuk** [comment]
I was able to fix the issue (it was in libuiohook, and was really about creating an invisible window). The fix will be available in the upcoming version 5.0.0, but I need to do some more stuff for it before releasing it. I'll close this issue when the new version is released.

**Comment by danielancines** [comment]
Hi, any idea when will be released? I'm using your awesome lib on a Production app and we need to dispose hook to improve the user experience.

Thank you.

**Comment by TolikPylypchuk** [comment]
I still have to do a couple more things to finalize the release, but I don't think it'll take long. I think in a couple weeks it should be ready (or even earlier - it depends on my free time really). Thanks for your interest!

**Comment by TolikPylypchuk** [comment]
I've just released version 5.0.0 which fixes this issue so it can be closed now.

**Comment by danielancines** [comment]
Nice job @TolikPylypchuk , I updated and everything is working fine. Thank you!
